widgets/labels/src/LabelsTable.jsx

```diff
@@ -0,0 +1,144 @@
+import _ from 'lodash';
+import { useState, useEffect } from 'react';
+import LabelValueInput from './LabelValueInput';
+import DeleteConfirmModal from './DeleteConfirmModal';
+
+export default function LabelsTable({ data, toolbox }) {
+    const { Button, DataTable, Icon } = Stage.Basic;
+    const { DeploymentActions, ManageLabelsModal } = Stage.Common;
+    const { useBoolean, useInput, useResettableState } = Stage.Hooks;
+
+    const [isAddModalOpen, openAddModal, closeAddModal] = useBoolean();
+    const [labelInEdit, setLabelInEdit, stopLabelEdit] = useResettableState();
+    const [currentLabelValue, setCurrentLabelValue] = useInput('');
+    const [labelToDelete, setLabelToDelete, unsetLabelToDelete] = useResettableState();
+    const [labels, setLabels] = useState(data.labels);
+
+    const actions = new DeploymentActions(toolbox);
+
+    useEffect(() => setLabels(data.labels), [JSON.stringify(data.labels)]);
+
+    function updateLabelValue() {
+        if (!currentLabelValue) return;
+
+        if (currentLabelValue === labelInEdit.value) {
+            stopLabelEdit();
+            return;
+        }
+
+        if (_.find(labels, { ...labelInEdit, value: currentLabelValue })) return;
+
+        labelInEdit.value = currentLabelValue;
+        setLabels([...labels]);
+        stopLabelEdit();
+        toolbox.loading(true);
+        actions.doSetLabels(data.deploymentId, labels).then(() => toolbox.loading(false));
+    }
+
+    function exportToJson() {
+        Stage.Utils.saveAs(
+            new Blob([JSON.stringify(labels.map(({ key, value }) => ({ [key]: value })))]),
+            `${data.deploymentId}-Labels.json`,
+            true
+        );
+    }
+
+    const hasManagePermission = Stage.Utils.isUserAuthorized('deployment_create', toolbox.getManagerState());
+
+    const tdStyle = { textOverflow: 'ellipsis', overflow: 'hidden' };
+
+    return (
+        <>
+            <DataTable
+                className="labelsTable"
+                totalSize={_.size(labels) > 0 ? undefined : 0}
+                noDataMessage="There are no Labels defined"
+            >
+                <DataTable.Column width="50%" label="Key" />
+                <DataTable.Column width="50%" label="Value" />
+                {hasManagePermission && <DataTable.Column width="80px" />}
+
+                {labels.map(item => (
+                    <DataTable.Row key={`${item.key}:${item.value}`} onClick={null}>
```

Comment:
I believe you no longer need the artificial `onClick` :slightly_smiling_face: It was fixed in ui-components

```suggestion
                    <DataTable.Row key={`${item.key}:${item.value}`}>
```

Reply:
Right, removed.